src/lib/registration/autoyast_addons.rb

```diff
@@ -64,9 +66,18 @@ def register
     def select_addons(all_addons)
       # select the requested addons from the AY profile
       requested_addons.each do |addon|
-        server_addon = all_addons.find do |a|
-          a.identifier == addon["name"] && a.version == addon["version"] &&
-            a.arch == addon["arch"]
+        # Set architecture if it is not defined in the requested addon
+        requested_arch = addon["arch"] || Yast::Arch.architecture
```

Comment:
So it could look like

```suggestion
        requested_arch = addon["arch"]
          || Y2Packager::ProductControlProduct::REG_ARCH[Yast::Arch.architecture]
          || Yast::Arch.architecture
```

And add `require "y2packager/product_control_product.rb"`.